### ISSUE TITLE
How are you adding the 2nd analog channel?

### ISSUE THREAD

**Issue by gregcrago** [dpavlin/DLO-138#2]
Are you purchasing a second DSO-138 board?
I have checked many forks and nobody is posting any pictures of this second channel hardware

**Comment by dpavlin** [comment]
I'm not original author, but @ardyesp has schematics for second channel https://github.com/ardyesp/DLO-138/blob/master/pics/HardwareMod.png

I guess that you have to build it, but I haven't tried it.

**Comment by gregcrago** [comment]
Yes, I have seen the 2nd channel schematic, but I have not seen any assembly pictures, either breadboard, or partially stuffed duplicate motherboard (DSO150)

**Comment by scimmiettarossa** [comment]
I did a simplified version on a breadboard, without capacitors, just to verify that everything is working, sending ta 1kHz test signal to both the native and additional channel

next step will be to implement it in a more structured manner, as soon as I receive the slide switches.. then i will post a picture

**Comment by scimmiettarossa** [comment]
created my version of the upgraded DLO138:
I've also added a slide switch to add the possibility to use also CHB as trigger, and another slide switch to enable a future mod to power supply it via USB or internal battery

Furthermore I've started to print a new extended case to contain the main and the extension board: this forced me to remove the small and hidden trigger LED, because with a non-transparent case it would be invisible, and to replace it with a new one with "high legs" to reach the top of the case

![P00613-212906](https://user-images.githubusercontent.com/32108707/84578592-fb54f400-adc6-11ea-8cdf-51855d9cadfc.jpg)
![Uploading P00613-212915.jpg…]()
![Uploading P00613-212955.jpg…]()
![Uploading P00613-213004.jpg…]()
![Uploading P00613-213215.jpg…]()
![Uploading P00613-215720.jpg…]()
![Uploading P00613-215822.jpg…]()

**Comment by ghost** [comment]
how to adjust the visible width of a wave, I see people use the switch but i didnt know this adjust by OPAMP or by code ?

**Comment by scimmiettarossa** [comment]
not sure I understand your question, but the only way I know to change the width of the wave, is by changing the timescale

**Comment by ghost** [comment]
Sorry! my english not well ==' .But many  Thanks for your help  ! 
so we change timescale(time base ) to change the width of wave
But we change the voltage scale to change the height of wave Right ? ( 1V/DIV -> wave display & 10mV/DIV -> wave display) -> so that my Q is : what is the principle make the change of wave display in 1V/DIV  & 10V/DIV  ? ( because hardware or code ?)

**Comment by scimmiettarossa** [comment]
both hw and code: the input range is selected via the sliding switches: as you can see in the schematic, they have dual contacts: the main one is used to "partition" the input voltage in a different way according to its range, while the secondary contact tells to the microcontroller which is the selected range, by sending to it a reference voltage that is different for each input level

**Comment by ghost** [comment]
So i have a question ?, when you add the channel 2 . The Switches of channel 2 is SW1_2 SW2_2 SW3_2 and so another side of Switches is connect to ?

**Comment by scimmiettarossa** [comment]
I did exactly as in the schematic from [https://raw.githubusercontent.com/dpavlin/DLO-138/encoder-swap/pics/HardwareMod.png](url): the new part is the one on top (outside the gray background, that is the base board schematic), and it is connected to the original board with the pink lines, and of course, with GROUND, AV+ and AV-, that you see in the last page of the manual [https://jyetech.com/wp-content/uploads/2018/07/dso138-user-manual-13803k-13804k.pdf](url) in the box "Power Supply"

**Comment by ghost** [comment]
What is V-MON used for ?
I saw in schematic have V-MON connect to PA6 pin but in code i not find where define V-MON, It only connect to resistor (10K and 1K ) with ( 79L05 ) to negative voltage
